PyPortal_Tides/admiralty_tides_graphical.py

```diff
@@ -0,0 +1,157 @@
+import time
+import board
+import displayio
+import json
+from secrets import secrets
+from adafruit_pyportal import PyPortal
+from adafruit_bitmap_font import bitmap_font
+from adafruit_display_text.label import Label
+
+#--| USER CONFIG |--------------------------
+STATION_ID = "0245"   # tide location, find yours from admiralty website
+PLOT_SIZE = 2            # tide plot thickness
+PLOT_COLOR = 0x00FF55    # tide plot color
+MARK_SIZE = 6            # current time marker size
+MARK_COLOR = 0xFF0000    # current time marker color
+DATE_COLOR = 0xE0CD1A    # date text color
+TIME_COLOR = 0xE0CD1A    # time text color
+VSCALE = 20              # vertical plot scale
+#-------------------------------------------
+
+# pylint: disable=line-too-long
+DATA_SOURCE = "https://admiraltyapi.azure-api.net/uktidalapi/api/V1/Stations/" + STATION_ID + "/TidalEvents?duration=1"
+DATA_LOCATION = []
+
+WIDTH = board.DISPLAY.width
+HEIGHT = board.DISPLAY.height
+
+# determine the current working directory needed so we know where to find files
+cwd = ("/"+__file__).rsplit('/', 1)[0]
+pyportal = PyPortal(url=DATA_SOURCE,
+                    headers={"Ocp-Apim-Subscription-Key":secrets['Ocp-Apim-Subscription-Key']},
+                    json_path=DATA_LOCATION,
+                    status_neopixel=board.NEOPIXEL,
+                    default_bg=cwd+"/tides_bg_graph.bmp")
+
+# Connect to the internet and get local time
+pyportal.get_local_time()
+
+# Setup palette used for plot
+palette = displayio.Palette(3)
+palette[0] = 0x0
+palette[1] = PLOT_COLOR
+palette[2] = MARK_COLOR
+palette.make_transparent(0)
+
+# Setup tide plot bitmap
+tide_plot = displayio.Bitmap(WIDTH, HEIGHT, 3)
+pyportal.splash.append(displayio.TileGrid(tide_plot, pixel_shader=palette))
+
+# Setup font used for date and time
+date_font = bitmap_font.load_font(cwd+"/fonts/mono-bold-8.bdf")
+date_font.load_glyphs(b'1234567890-')
+
+# Setup date label
+date_label = Label(date_font, text="0000-00-00", color=DATE_COLOR, x=7, y=14)
+pyportal.splash.append(date_label)
+
+# Setup time label
+time_label = Label(date_font, text="00:00:00", color=TIME_COLOR, x=234, y=14)
+pyportal.splash.append(time_label)
+
+# Setup current time marker
+time_marker_bitmap = displayio.Bitmap(MARK_SIZE, MARK_SIZE, 3)
+for pixel in range(MARK_SIZE * MARK_SIZE):
+    time_marker_bitmap[pixel] = 2
+time_marker = displayio.TileGrid(time_marker_bitmap, pixel_shader=palette, x=-MARK_SIZE, y=-MARK_SIZE)
+pyportal.splash.append(time_marker)
+
+def get_tide_data():
+    """Fetch JSON tide data and return parsed results in a list."""
+
+    # Get raw JSON data
+    raw_data = pyportal.fetch()
+    raw_data = json.loads(raw_data)
```

Comment:
Same comment as above.

Reply:
Returned a JSON array that caused an error. This resolved it